testing/BUILD.bazel

```diff
@@ -25,7 +25,11 @@ kt_android_library(
         "//third_party:androidx_test_ext_junit",
         "//third_party:androidx_test_runner",
         "//third_party:com_google_truth_truth",
+        "//third_party:com_squareup_retrofit2_converter-moshi",
+        "//third_party:com_squareup_retrofit2_retrofit",
+        "//third_party:com_squareup_retrofit2_retrofit-mock",
         "//third_party:junit_junit",
+        "//third_party:moshi",
```

Comment:
Added these four dependencies here, and tests started passing on Bazel now.

Reply:
That seems correct since RetrofitTestModule requires all of these. However, could we maybe inject the existing Retrofit coming from NetworkModule & add it to a new MockRetrofit provided by the new module? We aren't actually replacing the existing Retrofit binding, so it seems like that should "just work" & it will reduce code/dependency duplication here & in the test module.

Reply:
I didn't get this completely, don't we need a Retrofit object to initialize the Mockretrofit object in the RetrofitTestModule?

Reply:
I'm thinking that we could replace RetrofitTestModule with:

```kotlin
@Module
class RetrofitTestModule {
  @Provides
  @Singleton
  fun provideMockRetrofit(@OppiaRetrofit retrofit: Retrofit): MockRetrofit {
    val behavior = NetworkBehavior.create()
    behavior.setFailurePercent(0)

    return MockRetrofit.Builder(retrofit).apply {
      networkBehavior(behavior)
    }.build()
  }
}
```

And then we rely on NetworkModule to create & provide Retrofit. This simplifies dependencies and avoids needing to duplicate the Retrofit arrangement in both places (which is even more important if we change it--we want to keep both in sync).

Reply:
@BenHenning I tried this but this is showing error - 
![Screenshot from 2021-04-03 23-28-26](https://user-images.githubusercontent.com/54636525/113487219-744b4680-94d4-11eb-91b8-7d56d2d80f83.png)

I also changed the test file and it now looks like - 
```kotlin
@RunWith(AndroidJUnit4::class)
@LooperMode(LooperMode.Mode.PAUSED)
class ClassroomServiceTest {

  @Inject
  lateinit var retrofit: Retrofit

  @Inject
  lateinit var mockRetrofit: MockRetrofit

  @Before
  fun setUp() {
    setUpTestApplicationComponent()
  }

  @Test
  fun testClassroomService_usingFakeJson_deserializationSuccessful() {
    val delegate = mockRetrofit.create(ClassroomService::class.java)
    val mockClassroomService = MockClassroomService(delegate)

    val classroom = mockClassroomService.getClassroom("Math")
    val classroomResponse = classroom.execute()

    assertThat(classroomResponse.isSuccessful).isTrue()
    assertThat(classroomResponse.body()!!.topicSummaryDicts?.get(0)?.name).isEqualTo("Math")
  }

  private fun setUpTestApplicationComponent() {
    DaggerClassroomServiceTest_TestApplicationComponent
      .builder()
      .setApplication(ApplicationProvider.getApplicationContext()).build().inject(this)
  }

  // TODO(#89): Move this to a common test application component.
  @Singleton
  @Component(modules = [NetworkModule::class, RetrofitTestModule::class])
  interface TestApplicationComponent {
    @Component.Builder
    interface Builder {
      @BindsInstance
      fun setApplication(application: Application): Builder

      fun build(): TestApplicationComponent
    }

    fun inject(test: ClassroomServiceTest)
  }
}
```

Is the above code correct?

Reply:
@BenHenning I tried this but this is showing error - 
![Screenshot from 2021-04-03 23-28-26](https://user-images.githubusercontent.com/54636525/113487219-744b4680-94d4-11eb-91b8-7d56d2d80f83.png)